src/gepa/utils/stop_condition.py

```diff
@@ -0,0 +1,118 @@
+"""
+Utility functions for graceful stopping of GEPA runs.
+"""
+
+import os
+import time
+from typing import Callable
+
+
+class TimeoutStopCondition:
+    # stop callback that stops after a specified timeout
+    
+    def __init__(self, timeout_seconds: float):
+        self.timeout_seconds = timeout_seconds
+        self.start_time = time.time()
+    
+    def __call__(self) -> bool:
+        # return true if timeout has been reached
+        return time.time() - self.start_time > self.timeout_seconds
+
+
+class FileStopper:
```

Comment:
In api:optimize method, if log_dir is provided, create a FileStopper for "gepa.stop" automatically. Add this to optimize method's docstring.